### DISCUSSION TITLE
Support for Active Directory user lockout

### DISCUSSION THREAD

**Discussion by dteleguin** [keycloak/keycloak#21362] [Ideas]
Active Directory defines the [lockoutTime](https://learn.microsoft.com/en-us/windows/win32/adschema/a-lockouttime) attribute which is used to implement user lockout. When the user logs in via LDAP, there is no need for any special handling of that attribute at the Keycloak end (because LDAP bind would fail in that case).

However, if the user logs in using alternate methods (e.g. passkeys, FIDO etc.), the AD locked-out status won't be honored. Thus, if the user gets locked out via another channel, Keycloak login would still be possible.

Should we consider supporting the AD lockoutTime attribute natively?

**Comment by jmalyik** [comment]
In case of openldap, when one enables the ppolicy overlay, one can use the pwdAccountLockedTime for the same purpose. It would be really useful to do a general solution for detecting the lockout when there is an ldap federation configured in keycloak.

**Comment by mandree95** [comment]
In the context of passwordless passkeys, I believe this topic has become even more relevant. See, for example, the following issue: [!23069](https://github.com/keycloak/keycloak/issues/23069)
